javaparser-core/src/main/java/com/github/javaparser/ast/type/PrimitiveType.java

```diff
@@ -29,7 +31,32 @@
 public final class PrimitiveType extends Type {
 
 	public enum Primitive {
-		Boolean, Char, Byte, Short, Int, Long, Float, Double
+		Boolean ("Boolean"),
+		Char    ("Character"),
+		Byte    ("Byte"),
+		Short   ("Short"),
+		Int     ("Integer"),
+		Long    ("Long"),
+		Float   ("Float"),
+		Double  ("Double");
+
+		static final HashMap<String, PrimitiveType.Primitive> unboxMap = new HashMap<String, PrimitiveType.Primitive>();
+		static {
+			for(PrimitiveType.Primitive unboxedType : PrimitiveType.Primitive.values()) {
+				String boxedTypeName = unboxedType.toBoxedType().getName();
+				unboxMap.put(boxedTypeName, unboxedType);
+			}
+		}
+
+		private final ClassOrInterfaceType boxedType;
```

Comment:
You should directly store the name `String` here instead of the `ClassOrInterfaceType`. Indeed consider the following scenario:
```
ClassOrInterfaceType boxedInt = Primitive.Int.toBoxedType();
// Add a NonNull annotation
boxedInt.setAnnotations(/*...*/);
```
This would add annotations directly in the `ClassOrInterfaceType` that you store in `Primitive`. This is certainly not what a user would expect :p